src/SourceBuild/content/repo-projects/sdk.proj

```diff
@@ -26,8 +26,8 @@
     <BuildArgs>$(BuildArgs) /p:DOTNET_INSTALL_DIR=$(DotNetRoot)</BuildArgs>
 
     <BuildArgs Condition="'$(TargetOS)' != 'windows'">$(BuildArgs) /p:AspNetCoreInstallerRid=$(TargetRid)</BuildArgs>
-    <!-- sdk always wants to build portable on FreeBSD -->
-    <BuildArgs Condition="'$(TargetOS)' == 'freebsd' and '$(DotNetBuildSourceOnly)' == 'true'">$(BuildArgs) /p:PortableBuild=true</BuildArgs>
+    <!-- sdk always wants to build portable on FreeBSD etc. -->
+    <BuildArgs Condition="('$(TargetOS)' == 'freebsd' or '$(TargetOS)' == 'solaris') and '$(DotNetBuildSourceOnly)' == 'true'">$(BuildArgs) /p:PortableBuild=true</BuildArgs>
```

Comment:
Keeping it to solaris since illumos has distros https://github.com/dotnet/sdk/blob/9d273aebb5e984cfb9349ed0048231a2bca8fb0d/src/Cli/Microsoft.DotNet.Cli.Utils/RuntimeEnvironment.cs#L149